src/App.tsx

```diff
@@ -1,7 +1,36 @@
-import React from 'react';
+import { Component } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ key: event.key });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ key: event.key });
+    });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        {
+          this.state.key
+          // eslint-disable-next-line
+            ? <p className="App__message">The last pressed key is{`[${this.state.key}]`}</p>
+            : <p>Nothing was pressed yet</p>
+        }
+      </div>
+    );
+  }
+}
```

Comment:
the tests expect that element p is always present
thats why it fails
```suggestion
		const { key } = this.state;
		
    return (
      <div className="App">
        <p className="App__message">
          {key
            ? `The last pressed key is [${key}]`
            : 'Nothing was pressed yet'}
        </p>
      </div>
    );
  }
}
```